{
  try {
    p.protect();
  }
 catch (  AssertionFailedError e) {
    addFailure(test,e);
  }
catch (  ThreadDeath e) {
    throw e;
  }
catch (  Throwable e) {
    addError(test,e);
  }
}
